### PR TITLE
Timob 18854 : Add Delegate & Reply for openParentApplication to Titanium

### PR DESCRIPTION
**Pull request by benbahrenburg**
This PR relates to ticket [TIMOB-18854](https://jira.appcelerator.org/browse/TIMOB-18854)

Included in this PR is:

1.Addition of the required Application Delegate
2.Ti.App.iOS notification with information provided by the calling WatchKit Extension
3.sendWatchExtensionReply method to reply back to the [openParentApplication](https://developer.apple.com/library/ios/documentation/WatchKit/Reference/WKInterfaceController_class/#//apple_ref/occ/clm/WKInterfaceController/openParentApplication:reply:)

A detail of all the changes included in this PR is available [here](https://gist.github.com/benbahrenburg/0c9cfcd816fbd718fb08)

### CODE REVIEW

File: iphone/Classes/TiAppiOSProxy.m [hunk 1]
```diff
@@ -544,6 +551,39 @@ -(void)didRegisterUserNotificationSettingsNotification:(NSNotification*)notifica
...
+        kArgKey = 0,
+        kArgCount,
+        kArgUserInfo = kArgCount
+    };
+    
+    ENSURE_TYPE(args,NSArray);
+    ENSURE_ARG_COUNT(args, kArgCount);
+    
+    NSString *key = [TiUtils stringValue:[args objectAtIndex:kArgKey]];
+    NSDictionary *userInfo = [args objectAtIndex:kArgUserInfo];
```
Comment (pec1985) [inline iphone/Classes/TiAppiOSProxy.m]: The `kArgUserInfo` value is 1 so it should be inside the `if([args count] > 1)` block

**Comment by pec1985** [comment]
@benbahrenburg thanks for this PR. I left a minor comment that needs to be addressed in order to avoid crashes. 
One thing that's missing is documentation. Please look into the `apidoc/Titanium/App/iOS/iOS.yml`
Thanks!

**Comment by benbahrenburg** [comment]
@pec1985, you need to pass a nil in order to have it not crash. I would recommend reverting back to the initial submission.

**Comment by benbahrenburg** [comment]
@pec1985 documentation added.  Dictionary inlined.

**Comment by pec1985** [comment]
Closing PR, fixed a minor docuemtnation issue, new PR with @benbahrenburg's changes here: https://github.com/appcelerator/titanium_mobile/pull/6826

Thanks Ben!

**Pull request closed without merging**